app/lib/constituency_api.rb

```diff
@@ -0,0 +1,51 @@
+module ConstituencyApi
+  ConstituencyApiError = Class.new(RuntimeError)
+  
+  class Constituency
+    attr_accessor :name
+    
+    def initialize(name)
+      self.name = name
+    end
+    
+    def ==(another_constituency)
+      self.name == another_constituency.name
+    end
+  end
```

Comment:
It is excellent to have an actual data object, so +:100: for that!  As another bit of ruby knowledge we could write this as:

```ruby
class Constituency < Struct.new(:name)
end
```

And we'd get the initializer, accessor and equality for free.  It's worth exploring structs to see the pros-and-cons of using them.  Here's a starter blog post http://blog.steveklabnik.com/posts/2012-09-01-random-ruby-tricks--struct-new